core/src/main/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionRequestValidation.scala

```diff
@@ -63,23 +63,34 @@ object SubmissionRequestValidation extends StringValidationUtils {
   ).flatten
 
   def validateEntityNameAndType(submission: SubmissionRequest): Option[ErrorReport] =
-    if (submission.entityName.isDefined != submission.entityType.isDefined)
-      Some(
-        ErrorReport(
-          StatusCodes.BadRequest,
-          s"You must set both entityType and entityName to run on an entity, or neither (to run with literal or workspace inputs)."
+    if (submission.entityType.isDefined) {
+      if (submission.entityName.isDefined && submission.entityNames.isDefined) {
+        Some(
+          ErrorReport(
+            StatusCodes.BadRequest,
+            s"You must set either entityName or entityNames if running on an entity, but not both."
+          )
         )
-      )
-    else None
+      } else if (!submission.entityName.isDefined && !submission.entityNames.isDefined) {
+        Some(
+          ErrorReport(
+            StatusCodes.BadRequest,
+            s"You must set both entityType and entityName(s) to run on an entity, or neither (to run with literal or workspace inputs)."
+          )
+        )
+      } else None
+    } else None
 
+  // TODO make sure you've covered all cases correctly, this probably isn't right
   def validateMethodConfigRootEntity(submission: SubmissionRequest,
                                      methodConfig: MethodConfiguration
   ): Option[ErrorReport] =
-    (methodConfig.dataReferenceName, methodConfig.rootEntityType, submission.entityName) match {
-      case (Some(_), _, _)          => None
-      case (None, Some(_), Some(_)) => None
-      case (None, None, None)       => None
-      case (None, Some(_), None) =>
+    (methodConfig.dataReferenceName, methodConfig.rootEntityType, submission.entityName, submission.entityNames) match {
+      case (Some(_), _, _, _)          => None
+      case (None, Some(_), Some(_), _) => None
+      case (None, Some(_), _, Some(_)) => None
+      case (None, None, None, None)    => None
+      case (None, Some(_), None, None) =>
```

Comment:
yuck - at this point I'd rewrite this to be more readable. Just a thought:
```scala
      case (Some(_), _, _, _)          => None // data reference is defined
      case (None, rootEntityType, entityName, entityNames) =>
        if rootEntityType.isDefined && entityName.isEmpty && entityNames.isEmpty 
          // error 
        if rootEntityType.isEmpty && (entityName.isDefined || entityNames.isDefined)
          // error

      etc…
```

does this have unit tests? That could give confidence that you've covered all the cases.